nominal/core/_clientsbunch.py

```diff
@@ -64,3 +69,39 @@ def from_config(cls, cfg: ServiceConfiguration, agent: str, token: str) -> Self:
             upload=client_factory(upload_api.UploadService),
             video=client_factory(scout_video.VideoService),
         )
+
+    def auth_expiration(self) -> datetime.datetime:
+        """Returns the timestamp at which the current auth token becomes invalid
+
+        Raises:
+            ValueError: Malformed auth token present
```

Comment:
I get why the implementation here is more correct on a technical level, but from a simplicity perspective, maybe we should follow the same path as #82 

In this case, we shouldn't really need to check if it has expired, right? We can just catch the response error for a 401, and we'll "know" with sufficient confidence that the token has expired. especially because we validate that the original token is valid in the auth flow now